src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,8 +30,33 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String[] addressDetails = address.split(",");
+        if(addressDetails.length!=4){
+        	throw new IllegalValueException(MESSAGE_ADDRESS_WRONG_FORMAT);
+        }
+        unit = new Unit(addressDetails[0].trim());
+        block = new Block(addressDetails[1].trim());
+        street = new Street(addressDetails[1].trim());
+        postalcode = new PostalCode(addressDetails[1].trim());
```

Comment:
Doesn't look correct. Why the same value is used thrice? To prevent tests breaking? :question: